File: app/src/main/java/com/example/projetoestagioaiko/api/OlhoVivoApiService.java
package com.example.projetoestagioaiko.api;


import com.example.projetoestagioaiko.models.LinhaOnibus;
import com.example.projetoestagioaiko.models.PontoDeParada;
import com.example.projetoestagioaiko.models.PontosDeParadaResponse;
import com.example.projetoestagioaiko.models.PrevisaoChegada;
import com.example.projetoestagioaiko.models.Veiculo;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface OlhoVivoApiService {
    @POST("Login/Autenticar")
    Call<Boolean> autenticar(@Query("token") String token);

    @GET("Parada/Buscar")
    Call<List<PontoDeParada>> buscarParadas(@Query("termosBusca") String termosBusca);

    @GET("Parada/BuscarParadasPorLinha")
    Call<List<PontoDeParada>> buscarParadasPorLinha(@Query("codigoLinha") int codigoLinha);

    @GET("Posicao")
    Call<List<Veiculo>> obterPosicoesVeiculos();

    @GET("Linha/Buscar")
    Call<List<LinhaOnibus>> obterLinhas();

    @GET("Parada/Buscar")
    Call<PontosDeParadaResponse> obterPontosDeParada();

    @GET("Previsao/Linha")
    Call<List<PrevisaoChegada>> obterPrevisaoChegada(@Query("codigoLinha") String LinhaId);

    @GET("Previsao/Parada")
    Call<List<PrevisaoChegada>> obterPrevisaoChegadaPorParada(@Query("codigoParada") String paradaId);


}

File: app/src/main/java/br/vino/transmobisp/service/olho_vivo/OlhoVivoApiService.kt
package br.vino.transmobisp.service.olho_vivo

import br.vino.transmobisp.model.ArrivalForecastResponse
import br.vino.transmobisp.model.Line
import br.vino.transmobisp.model.Stop
import br.vino.transmobisp.model.VehicleResponse
import br.vino.transmobisp.model.stops_from_line.StopsFromLine
import br.vino.transmobisp.model.vehicles_lines_from_stop.StopWithLinesAndVehicles
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

interface OlhoVivoApiService {
    @POST("Login/Autenticar")
    suspend fun authenticate(@Query("token") token: String): Response<Boolean>

    @GET("Posicao")
    suspend fun getVehicles(): Response<VehicleResponse>

    @GET("Parada/BuscarParadasPorLinha")
    suspend fun getStops(@Query("codigoLinha") lineCode: String): Response<List<Stop>>

    @GET("Previsao/Linha")
    suspend fun getStopsFromLine(@Query("codigoLinha") lineCode: String): Response<StopsFromLine>

    @GET("Parada/Buscar")
    suspend fun getStopWithTerm(@Query("termosBusca") lineCode: String): Response<List<Stop>>

    @GET("Previsao/Parada")
    suspend fun getLinesWithVehiclesFromStop(@Query("codigoParada") stopCode: String): Response<StopWithLinesAndVehicles>
}

